pylint/checkers/variables.py

```diff
@@ -2074,6 +2055,29 @@ def _is_variable_violation(
 
         return maybe_before_assign, annotation_return, use_outer_definition
 
+    @staticmethod
+    def _maybe_used_and_assigned_at_once(defstmt: nodes.Statement) -> bool:
+        """Check if `defstmt` has the potential to use and assign a name in the
+        same statement.
+        """
+        if isinstance(defstmt.value, nodes.BaseContainer) and defstmt.value.elts:
+            # The assignment must happen as part of the first element
+            # e.g. "assert x:= True, x"
+            # NOT "assert x, x:= True"
+            value = defstmt.value.elts[0]
+        else:
+            value = defstmt.value
+        if isinstance(value, nodes.IfExp):
+            return True
+        if isinstance(value, nodes.Lambda) and isinstance(value.body, nodes.IfExp):
+            return True
+        if isinstance(value, nodes.Call) and (
+            any(isinstance(kwarg.value, nodes.IfExp) for kwarg in value.keywords)
+            or any(isinstance(arg, nodes.IfExp) for arg in value.args)
+        ):
+            return True
+        return False
```

Comment:
We could return directly the content of the if, but I guess you did it that way for better extendability.
```suggestion
        return isinstance(value, nodes.Call) and (
            any(isinstance(kwarg.value, nodes.IfExp) for kwarg in value.keywords)
            or any(isinstance(arg, nodes.IfExp) for arg in value.args)
        )
```